libs/common/src/lib/resources/resource-list/resource-list.component.ts

```diff
@@ -1,125 +1,219 @@
+  get isErrorView(): boolean {
+    return this.resourceListService.status === 'ERROR';
   }
 
-  standalone = this.sdk.nuclia.options.standalone;
-  emptyKb = this.resourceListService.emptyKb;
-  isTrial = this.features.isTrial;
-  isAccountManager = this.features.isAccountManager;
-  upgradeUrl = this.sdk.currentAccount.pipe(map((account) => this.navigation.getUpgradeUrl(account.slug)));
-
   constructor(
     private sdk: SDKService,
-    private translate: TranslateService,
     private uploadService: UploadService,
-    private navigation: NavigationService,
     private resourceListService: ResourceListService,
     private tracking: STFTrackingService,
     private features: FeaturesService,
-  ) {}
-
-  ngOnInit(): void {
-    // Reset resource list when query is empty (without forcing user to hit enter)
-    this.searchForm.controls.query.valueChanges
+    private route: ActivatedRoute,
+    private router: Router,
+    private cdr: ChangeDetectorRef,
+  ) {
+    this.uploadService.updateStatusCount().subscribe(() => {
+      // updateStatusCount is launching the first `/catalog` request which will set the shard.
+      // we need to wait for it to be done before launching other request to prevent using different shards for different requests
+      this.isShardReady.next(true);
+    });
+
+    this.ready
       .pipe(
-        debounceTime(100),
-        filter((value) => !value),
+        distinctUntilChanged(),
+        filter((ready) => ready),
+        switchMap(() => {
+          if (this.isMainView || this.isProcessedView) {
+            return this.loadFilters();
+          } else {
+            this.filterOptions = { classification: [], mainTypes: [], creation: {} };
+            this.cdr.markForCheck();
+            return of(null);
+          }
+        }),
         takeUntil(this.unsubscribeAll),
       )
-      .subscribe(() => this.search());
+      .subscribe();
 
-    // we need to wait for the translations to be loaded before setting the search options
-    // because the default value (=title) is used to fill in the select input and this value is not refreshed
-    // when the options are provided as ngContent
-    this.translate
-      .stream(['resource.search.in-title', 'resource.search.in-resource'])
+    merge(this.resourceListService.ready, this.resourceListService.filters, this.uploadService.statusCount)
       .pipe(takeUntil(this.unsubscribeAll))
-      .subscribe((trans) => {
-        this.searchOptions = [
-          new OptionModel({ id: 'title', label: trans['resource.search.in-title'], value: 'title' }),
-          new OptionModel({
-            id: 'resource',
-            label: trans['resource.search.in-resource'],
-            value: 'resource',
-          }),
-        ];
+      .subscribe(() => {
+        this.resourceListService.setHeaderHeight(this.header?.nativeElement.clientHeight || 0);
       });
   }
 
   ngOnDestroy() {
+    this.isShardReady.next(false);
```

Comment:
Maybe that's a bit radical (as the shard is still valid after leaving the resource list page, but I understand we need an observable anyway because when entering another kb, we have no shard anymore). I guess we could probably expose the `Nuclia.currentShards` attribute as an observable, but that will involve some refactoring. Let's keep it that way for now.